base/bucket_n1ql_test.go

```diff
@@ -40,12 +41,21 @@ func TestN1qlQuery(t *testing.T) {
 	indexExpression := "val"
 	err := bucket.CreateIndex("testIndex_value", indexExpression, "", testN1qlOptions)
 	if err != nil {
-		t.Errorf("Error creating index: %s", err)
+		if !strings.Contains(err.Error(), "index testIndex_value already exists") {
```

Comment:
This feels brittle, but I realize it's something from another module that might be out of our control.  Do you think it's worth trying to file an issue to get some sort of error codes?  Also, in the meantime what about at least putting this into a constant or centralized function in case other code needed the same check?

Eg something like: `function IsErrorIndexAlreadyExists(err error) bool { .. } `

Reply:
Oh, I just realized this was test code .. so I guess it matters a bit less.

Reply:
Agreed - see https://github.com/couchbase/sync_gateway/pull/3476/files#diff-040de30dacf175a84e9906a8203de520R262